src/main/java/com/shipmentEvents/demo/EventHandler.java

```diff
@@ -52,10 +59,24 @@
             return "SUCCESS";
         } catch (final Exception ex) {
             logger.log(String.format("Failed to process shipment Updates in %s due to %s", scheduledEvent.getAccount(), ex.getMessage()));
-            throw new RuntimeException(ex);
+            throw new RuntimeException("Hiding the exception");
         }
     }
 
+    public String weakMessageEncryption(String message, String key) throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like your code uses a cipher object with an insecure transformation. To make your code more secure, use one of the following algorithms with a built-in integrity check: `AES/GCM/NoPadding` or `ChaCha20-Poly1305`.

[Learn more](https://cwe.mitre.org/data/definitions/327.html)